pennylane/grouping/pauli.py

```diff
@@ -0,0 +1,254 @@
+
+    The group can also be created using a custom wire map (if no map is
+    specified, a default map of label :math:`i` to wire ``i`` will be created).
+
+    .. code-block:: python
+
+        n_qubits = 3
+        wire_map = {'a' : 0, 'b' : 1, 'c' : 2}
+
+        for p in pauli_group(n_qubits, wire_map=wire_map):
+            print(p)
+
+    """
+    if not isinstance(n_qubits, int):
+        raise TypeError("Must specify an integer number of qubits construct the Pauli group.")
+
+    if n_qubits <= 0:
+        raise ValueError("Number of qubits must be at least 1 to construct Pauli group.")
+
+    return _pauli_group_generator(n_qubits, wire_map=wire_map)
+
+
+def pauli_mult(pauli_1, pauli_2, wire_map=None):
+    """Multiply two Pauli words together.
+
+    Two Pauli operations can be multiplied together by taking the additive
+    OR of their binary symplectic representations.
+
+    Args:
+        pauli_1 (.Operation): A Pauli word.
+        pauli_2 (.Operation): A Pauli word to multiply with the first one.
+        wire_map (dict[Union[str, int], int]): dictionary containing all wire labels used in the Pauli
+            word as keys, and unique integer labels as their values. If no wire map is
+            provided, the map will be constructed from the set of wires acted on
+            by the input Pauli words.
+
+    Returns:
+        .Operation: The product of pauli_1 and pauli_2 as a Pauli word
+            (ignoring the global phase).
+
+    **Example**
+
+    This function enables multiplication of Pauli group elements at the level of
+    Pauli words, rather than matrices. For example,
+
+    >>> from pennylane.pauli import pauli_mult
+    >>> pauli_1 = qml.PauliX(0) @ qml.PauliZ(1)
+    >>> pauli_2 = qml.PauliY(0) @ qml.PauliZ(1)
+    >>> product = pauli_mult(pauli_1, pauli_2)
+    >>> print(product)
+    PauliZ(wires=[0])
+    """
+
+    # If no wire map is specified, generate one from the union of wires
+    # in both Paulis.
+    if wire_map is None:
+        wire_labels = set(pauli_1.wires.labels + pauli_2.wires.labels)
+        wire_map = {label: i for i, label in enumerate(wire_labels)}
+
+    # Check if pauli_1 and pauli_2 are the same; if so, the result is the Identity
+    if are_identical_pauli_words(pauli_1, pauli_2):
+        first_wire = list(wire_map.keys())[0]
+        return Identity(first_wire)
+
+    # Compute binary symplectic representations
+    pauli_1_binary = pauli_to_binary(pauli_1, wire_map=wire_map)
+    pauli_2_binary = pauli_to_binary(pauli_2, wire_map=wire_map)
+
+    bin_symp_1 = np.array([int(x) for x in pauli_1_binary])
+    bin_symp_2 = np.array([int(x) for x in pauli_2_binary])
+
+    # Shorthand for bitwise XOR of numpy arrays
+    pauli_product = bin_symp_1 ^ bin_symp_2
+
+    return binary_to_pauli(pauli_product, wire_map=wire_map)
+
+
+def pauli_mult_with_phase(pauli_1, pauli_2, wire_map=None):
+    r"""Multiply two Pauli words together including the global phase.
```

Comment:
what does 'including' mean in this context? As in, 'taking into account the global phase'?

Reply:
Ohhh, as in, "Multiply two Pauli words together and return both the product and the global phase."

Reply:
Good catch, will clarify that in both docstrings!